cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/distributed/AbstractDistributedProgramRunner.java

```diff
@@ -183,6 +184,10 @@ public final ProgramController run(final Program program, final ProgramOptions o
       final ProgramOptions options = addArtifactPluginFiles(oldOptions, localizeResources,
                                                             DirUtils.createTempDir(tempDir));
 
+      // Add extra jars set in cConf to additionalClassPaths and localizeResources
+      LocalizationUtils.addExtraJars(localizeResources, cConf);
+      final List<String> additionalClassPaths = LocalizationUtils.getExtraJarNames(cConf);
```

Comment:
not good to call two different methods to extra jar file and jar names and assume they'll match. Better use the result of `addExtraJars` to setup addition classpath (meaning making `addExtraJars` returning a `Map<String, LocalizeResource>` for the set of resources it added).